syncback/krsync.sh

```diff
@@ -29,19 +35,20 @@ install_rsync_tilt() {
     set -e
     if [[ "$exit_code" != "0" ]]; then
         # shellcheck disable=SC2002
+        echo "installing tilt's rsync"
         cat "$KRSYNC_TAR_PATH" | kubectl exec -i $K8S_OBJECT -- tar -xf - -C $(dirname $rsync_tilt_path)
     fi
 }
 
 find_rsync_path() {
     local rsync_path=--rsync-path=$rsync_tilt_path
-    # ensure pipeline status is non-zero if rsync is missing in the container
-    set -o pipefail
     # see if we already have a compatible rsync on the container
-    detected_rsync_version=$(kubectl exec $K8S_OBJECT -- rsync --version 2> /dev/null \
-                              | head -1 | awk '{ print $3 }')
-    if [ $? -eq 0 -a "$detected_rsync_version" ] && [[ "$detected_rsync_version" > 3 ]]; then
-        rsync_path=
+    rsync_output=$(kubectl exec $K8S_OBJECT -- rsync --version 2> /dev/null)
```

Comment:
In [this previous run](https://app.circleci.com/pipelines/github/tilt-dev/tilt-extensions/943/workflows/037f4dfe-a1cf-4a94-afa8-372189868715/jobs/2088?invite=true#step-103-2135) it looks like kubectl may have returned a non-zero exit code even though it looks like rsync wrote output to the pipe. I wonder if we should disregard kubectl exit status entirely and go back to do something like 
```
detected_rsync_version=$(kubectl -v=0 exec $K8S_OBJECT -- rsync --version 2> /dev/null | grep rsync | head -1 | awk '{ print $3 }')
if [ "$detected_rsync_version" ] && [[ "$detected_rsync_version" > 3 ]]; then
```

Reply:
or 
```
detected_rsync_version=$(kubectl -v=0 exec $K8S_OBJECT -- rsync --version 2> /dev/null | grep "rsync *version" | awk '{ print $3 }')
```

Reply:
ya, the previous run is from before the previous PR, that didn't have the pipe changes.

i'm not totally convinced the grep solutions will work right, since they have the same pipe problems.